Update backend local development docs

SuaYoo opened the pull request:
- Minor adjustments to [Local Deployment](https://docs.browsertrix.cloud/deploy/local/) guide to clarify steps needed for local development vs. local trial deployment.
- Adds "Running the Backend" page with mostly links back to "Local Deployment", with callouts to dev-specific configs. ([Markdown preview for easy review](https://github.com/webrecorder/browsertrix-cloud/tree/docs-update-local-deployment/docs/develop/backend-dev.md))

Related to https://github.com/webrecorder/browsertrix-cloud/issues/881

SuaYoo committed:
clarify local deployment

SuaYoo committed:
add local deployment guide

SuaYoo committed:
update admin instructions

SuaYoo committed:
update username

SuaYoo commented on docs/deploy/local.md:
```diff
@@ -127,31 +122,28 @@ By default, the database + storage volumes are not automatically deleted, so you
-    MicroK8s uses its own container registry, running on port 32000. 
+    MicroK8s uses its own container registry, running on port 32000.
 
-    1. Set `export REGISTRY=localhost:32000/` and then run `./scripts/build-backend.sh` and/or `./scripts/build-frontend.sh` to rebuild the images into the MicroK8S registry. 
```
Oops, some whitespace formatting changes made it in.

Shrinks99 replied:
Happens to the best of us 🙃

tw4l commented on docs/deploy/local.md:
```diff
@@ -66,14 +64,11 @@ helm upgrade --install -f ./chart/values.yaml -f ./chart/examples/local-config.y
 
-Note that the admin user and password will not be reset after creation.
-
+For example, to enable a default superadmin user, uncomment the `superadmin` block in `local-config.yaml`. An admin user with username `admin@example.com` and password `PASSW0RD!` will be automatically created. Note that the admin user and password will not be reset after creation.
```
AFAIK, the default superuser will be created from the values in `values.yaml`, and it's only necessary to uncomment/modify in `local-config.yaml` if you want to override the defaults.

tw4l commented on docs/develop/backend-dev.md:
````diff
@@ -0,0 +1,58 @@
+  # optional: if not set, automatically generated
+  # change or remove this
+  password: PASSWORD!
+```
````
Same comment as above, I think we can either remove this or change the text to "Uncomment `superuser` to override the default superadmin admin credentials".

ikreymer replied:
Yea, maybe can say for anything other than local testing, should override

tw4l commented on docs/deploy/local.md:
```diff
@@ -66,14 +64,11 @@ helm upgrade --install -f ./chart/values.yaml -f ./chart/examples/local-config.y
 
 The local setup includes the full Browsertrix Cloud system, with frontend, backend api, db (via MongoDB) and storage (via Minio)
 
-An admin user with name `admin@example.com` and password `PASSW0RD!` will be automatically created.
```
This should be true! And is working for me, without needing to uncomment anything locally.

tw4l commented on docs/develop/backend-dev.md:
````diff
@@ -0,0 +1,58 @@
+cp ./chart/examples/local-config.yaml ./chart/local.yaml
+```
+
+Changes to `local.yaml` will not be tracked in git.
````
Thank you for adding this! 🙌

tw4l commented on docs/develop/backend-dev.md:
```diff
@@ -0,0 +1,58 @@
+# Running the Backend
```
This is a great addition, thank you!

tw4l reviewed (commented):
Thanks! Left a few small comments.

Shrinks99 commented on docs/develop/backend-dev.md:
```diff
@@ -0,0 +1,58 @@
+
+## Configuration
+
+For quick setup, we've provided a Helm chart configuration file `examples/local-config.yaml`, which you can copy as `local.yaml`. From the root directory:
```
```suggestion
For quick setup, we provide a Helm chart configuration file `examples/local-config.yaml`, which you can copy as `local.yaml`. From the root directory:
```

Might switch to present tense?

ikreymer committed:
docs refactor:
- move local dev setup specific docs to local-dev-setup.md
- deploy/local.md focuses only on deployment with latest release
- add nav to mkdocs.yml to ensure correct order of pages

ikreymer committed:
text tweak

tw4l commented on docs/develop/local-dev-setup.md:
````diff
@@ -0,0 +1,118 @@
+        frontend_image: "localhost:32000/webrecorder/browsertrix-frontend:latest"
+        ```
+
+        Note: whenever `helm` is mentioned, you'll need to run `microk8s helm3` instead.
````
Maybe also add a note here that the pull policies should be:

```yaml
ackend_pull_policy: 'IfNotPresent'
frontend_pull_policy: 'IfNotPresent'
```

ikreymer replied:
Ah, right, nice catch. Added it in previous section on pull policies.

tw4l replied:
Nice!

tw4l commented:
Refactor looks great! Just one comment related to microk8s

ikreymer committed:
add microk8s specific info

ikreymer committed:
more doc tweaks for Minikube

ikreymer committed:
more tweaks for minikube
